shipstation_integration/shipstation_integration/doctype/shipstation_settings/shipstation_settings.js

```diff
@@ -11,6 +11,36 @@ function company_query(frm, cdt, cdn) {
   };
 }
 
+function set_item_field_options(frm) {
+  let options_arr = [];
+  delete locals.DocType['Sales Order Item'];
+  // This loads the "Sales Order Item" doctype meta into locals
+  frappe.model.with_doctype("Sales Order Item", () => {
+    const fieldlist = frappe.get_meta("Sales Order Item").fields;
```

Comment:
Both of these first two lines work together. `frappe.model.with_doctype("Sales Order Item", () => {` loads the doctype into the locals on the current session. The reason we run `delete locals.DocType['Sales Order Item'];` is to force a refresh of the doctype meta, in case custom fields were added or removed since the last run.

Reply:
> A better way might be to just fetch the fields from the server directly.

The approach above seems to work great, in my experience. The big challenge with this field list is that it needs to have a combined list of both built-in docfields and custom fields. This is the most elegant solution I found to address this need, but if you have another suggestion, please do tell.

Reply:
Oh, that makes sense. Somehow I completely missed that `with_doctype` call.